### PR TITLE
Profile page improvements added

### PR DESCRIPTION
**Pull request by arfakbaba**
* implemented public profile 
* implemnted private profile 
* implemented edit profile backend connections

### CODE REVIEW

File: app/client/pages/user/[user_id].js [hunk 1]
```diff
@@ -12,13 +12,17 @@ import {
...
   TextField,
+  InputLabel
 } from '@mui/material';
+import * as Yup from "yup";
 import { API_URL } from "../../next.config";
 import axios from 'axios';
 import { useEffect, useState } from "react";
 import { useRouter } from 'next/router'
 import UserLayout from '../../layouts/user-layout/UserLayout';
+import { authorization } from '../../../server/src/services/auth';
```
Comment (nurlandadashov02) [inline app/client/pages/user/[user_id].js:23]: remove this we should not access backend code in frontend

File: app/client/pages/user/[user_id].js [hunk 2]
```diff
@@ -110,7 +152,7 @@ export default function profile(props) {
                     }}
                   >
                     <Avatar
-                      src={post.avatar}
+                      src={personalValues.image}
```
Comment (nurlandadashov02) [inline app/client/pages/user/[user_id].js:155]: `${API_URL}/user/${personalValues.image}`. image url should be in this form. Correct this and make sure that profile photo is visible

File: app/client/pages/user/[user_id].js [hunk 3]
```diff
@@ -273,11 +339,369 @@ export default function profile(props) {
...
+                        variant="outlined"
+                      />
+                    </Grid>
+                    <Grid
+                      item
+                      md={12}
+                      xs={12}
+                    >
+                      <InputLabel shrink>
+                        Knowledges
```
Comment (nurlandadashov02) [inline app/client/pages/user/[user_id].js:648]: Knowledge

File: app/client/pages/user/[user_id].js [hunk 4]
```diff
@@ -242,10 +306,10 @@ export default function profile(props) {
                         <TextField
                           fullWidth
                           label="Intrests"
```
Comment (nurlandadashov02) [inline app/client/pages/user/[user_id].js:308]: Interests

File: app/client/pages/user/[user_id].js [hunk 5]
```diff
@@ -242,10 +306,10 @@ export default function profile(props) {
                         <TextField
```
Comment (nurlandadashov02) [inline app/client/pages/user/[user_id].js:306]: Knowledge and interests should be tag based components. You can use the code here: https://stackoverflow.com/questions/71585321/create-tag-using-autocomplete-material-ui-when-user-input-anything.   When viewing public profile disable the input

**Review by nurlandadashov02 — CHANGES_REQUESTED** [review]
Good job. But please correct points I made in comments. Thanks.

**Review by nurlandadashov02 — CHANGES_REQUESTED** [review]
You have to make interests and knowledge as tag input in edit mode as well. Update profile backend connection is not working properly. Make sure you are providing correct payload to request. If problem continues, contact backend team.

**Review by nurlandadashov02 — APPROVED** [review]

**Review by berkeozdemir2016400246 — APPROVED** [review]

**Pull request merged**